langchain/callbacks/arize_callback.py

```diff
@@ -1,217 +1,314 @@
 
-    def on_tool_start(
+    def _add_to_chain_data_on_chain_end(self, output: Dict[str, Any]) -> None:
+        if is_retrieval_augmented_generation_data(self._chain_data):
+            self._chain_data.response_text = output["result"]
+
+    def on_retriever_start(
         self,
         serialized: Dict[str, Any],
-        input_str: str,
+        query: str,
+        *,
+        run_id: UUID,
+        parent_run_id: Optional[UUID] = None,
+        tags: Optional[List[str]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
         **kwargs: Any,
-    ) -> None:
-        pass
+    ) -> Any:
+        ...
+
+    def on_retriever_end(
+        self,
+        documents: Sequence[LangChainDocument],
+        *,
+        run_id: UUID,
+        parent_run_id: Optional[UUID] = None,
+        **kwargs: Any,
+    ) -> Any:
+        if is_retrieval_augmented_generation_data(self._chain_data):
+            # TODO: we need document ids and cosine similarity scores here
+            pass
+
+    @abstractmethod
+    def _log_chain_data(self, chain_data: ChainData) -> None:
+        ...
 
-    def on_agent_action(self, action: AgentAction, **kwargs: Any) -> Any:
-        """Do nothing."""
-        pass
+    @staticmethod
+    def _is_base_chain(parent_run_id: Optional[UUID]) -> bool:
+        return parent_run_id is None
 
-    def on_tool_end(
+
+class ArizeCallbackHandler(BaseArizeCallbackHandler):
+    arize = _import_package("arize")
+
+    def __init__(
         self,
-        output: str,
-        observation_prefix: Optional[str] = None,
-        llm_prefix: Optional[str] = None,
+        arize_client: "ArizeClient",
+        model_id: str,
+        model_version: str,
+        *args: Any,
         **kwargs: Any,
     ) -> None:
-        pass
+        validate_package_version(self.arize, None, None)
+        super().__init__(**kwargs)
+        self._arize_client = arize_client
+        self._model_id = model_id
+        self._model_version = model_version
+
+    @classmethod
+    def from_credentials(
+        cls,
+        api_key: str,
+        space_key: str,
+        model_id: str,
+        model_version: str,
+        arize_client_kwargs: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> "ArizeCallbackHandler":
+        ArizeClient = import_arize_client()
+        arize_client = ArizeClient(
+            space_key=space_key,
+            api_key=api_key,
+            **(arize_client_kwargs or {}),
+        )
+        return cls(arize_client, model_id, model_version, **kwargs)
+
+    def _log_chain_data(self, chain_data: ChainData) -> None:
+        # TODO: implement logging of chain data to arize
+        print(self._chain_data)
+
 
-    def on_tool_error(
-        self, error: Union[Exception, KeyboardInterrupt], **kwargs: Any
+class PhoenixCallbackHandler(BaseArizeCallbackHandler):
+    pandas = _import_package("pandas")
+    phoenix = _import_package("phoenix")
+    pyarrow = _import_package("pyarrow")
```

Comment:
this doesn't belong in the arize callback handler IMO. And as mentioned above, phoenix is a runtime - it can't be imported like this just to run the callback.